allennlp/data/vocabulary.py

```diff
@@ -285,6 +287,48 @@ def from_dataset(cls,
                           max_vocab_size=max_vocab_size,
                           non_padded_namespaces=non_padded_namespaces)
 
+    @classmethod
+    def from_params(cls, params: Params, dataset=None):
+        """
+        There are two possible ways to build a vocabulary; from a
+        pre-existing dataset, using :func:`Vocabulary.from_dataset`, or
+        from a pre-saved vocabulary, using :func:`Vocabulary.from_files`.
+        This method wraps both of these options, allowing their specification
+        from a ``Params`` object, generated from a JSON configuration file.
+
+        Parameters
+        ----------
+        params: Params, required.
+        dataset: Dataset, optional.
+            If ``params`` doesn't contain a ``vocabulary_directory`` key,
+            the ``Vocabulary`` can be built directly from a ``Dataset``.
+
+        Returns
+        -------
+        A ``Vocabulary``.
+        """
+        vocabulary_directory = params.pop("vocabulary_directory", None)
```

Comment:
Remember, this will be under `{"vocabulary": {}}`, so you don't need to say "vocabulary" again.  I'd have this key be something like `from_directory`, or `from_files` instead.  It makes it a little more obvious what's going to happen when you give that key, too.

Reply:
sure.